.github/workflows/unit_tests.yml

```diff
@@ -0,0 +1,49 @@
+name: 'unit_tests'
+
+on:
+  pull_request:
+
+env:
+  TARGETS: f7
+  DEFAULT_TARGET: f7
+
+jobs:
+  main:
+    runs-on: [self-hosted, FlipperZeroTest]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: 'Get flipper from device manager (mock)'
+        id: device
+        run: |
+          echo "flipper=/dev/ttyACM0" >> $GITHUB_OUTPUT
+
+      - name: 'Compile unit tests firmware'
+        id: compile
+        continue-on-error: true
+        run: |
+          FBT_TOOLCHAIN_PATH=/opt ./fbt FIRMWARE_APP_SET=unit_tests flash_usb FORCE=1
```

Comment:
flash_usb will not flash with flipper if it hangs in case of critical error. You need to use a separate programmer for the flashing process.